File: Nokia Menu/NokiaMenuMap.py

    print("           NOKIA 3310 PHONE MENU")

    menus = ["PhoneBook", "Messages", "Chat", "Call register"]
    phoneBook = ["Search", "Service Nos.", "Add name", "Erase", "Options"]
    options = ["Type of view", "Memory status"]
    messages = ["Write message", "Inbox", "Outbox", "Message settings"]
    messageSettings = ["Set 1", "Common"]
    messageSet1 = ["Message centre number", "Message sent as", "Message validity"]
    messageCommon = ["Delivery reports", "Reply via same centre", "Character support"]
    callRegister = ["Missed calls", "Received calls", "Dialled numbers", "Erase recent call lists", "Show call duration"]
    callDuration = ["Last call duration", "All calls' duration", "Received calls' duration", "Dialled calls' duration", "Clear timers"]

   
    for i in range(len(menus)):
        print(f"{i+1} {menus[i]}")
    print("0 Exit")

    pick = int(input("Pick Menu (0-4): "))

    if pick == 1:  # PhoneBook
        for j in range(len(phoneBook)):
            print(f"{j+1} {phoneBook[j]}")
        print("0 Back")

        choice = int(input("Choice (0-5): "))

        if choice in range(1, 6):
            print(phoneBook[choice-1])

            if choice == 5:  # Options
                for k in range(len(options)):
                    print(f"{k+1} {options[k]}")
                print("0 Back")

                click = int(input("Click (0-2): "))
                if click in (1, 2):
                    print(options[click-1])
                elif click == 0:
                    print("Back to PhoneBook Menu")
        elif choice == 0:
            print("Back to Main Menu")

    elif pick == 2:  # Messages
        for j in range(len(messages)):
            print(f"{j+1} {messages[j]}")
        print("0 Back")

        msgChoice = int(input("Choice (0-4): "))

        if msgChoice == 4:  # Message settings
            for m in range(len(messageSettings)):
                print(f"{m+1} {messageSettings[m]}")
            print("0 Back")

            setChoice = int(input("Choice (0-2): "))
            if setChoice == 1:
                for s in range(len(messageSet1)):
                    print(f"{s+1} {messageSet1[s]}")
            elif setChoice == 2:
                for c in range(len(messageCommon)):
                    print(f"{c+1} {messageCommon[c]}")
            elif setChoice == 0:
                print("Back to Messages Menu")
        elif msgChoice in (1, 2, 3):
            print(messages[msgChoice-1])
        elif msgChoice == 0:
            print("Back to Main Menu")

    elif pick == 3:  # Chat
        print(menus[2])
        print("0 Back")
        input()

    elif pick == 4:  # Call register
        for j in range(len(callRegister)):
            print(f"{j+1} {callRegister[j]}")
        print("0 Back")

        regChoice = int(input("Choice (0-5): "))

        if regChoice in range(1, 5):
            print(callRegister[regChoice-1])
        elif regChoice == 5:  # Show call duration
            for d in range(len(callDuration)):
                print(f"{d+1} {callDuration[d]}")
            print("0 Back")
            input()
        elif regChoice == 0:
            print("Back to Main Menu")

    elif pick == 0:
        print("Exiting...")

    else:
        print("Invalid choice!")

File: Nokia Menu/NokiaMenuMap.java
import java.util.Scanner;

public class NokiaMenuMap {
   public static void main(String[] args) {
      Scanner input = new Scanner(System.in);

String[] menus = {"PHONE BOOK", "MESSAGES", "CHAT", "CALL REGISTER", "TONES","SETTINGS", "CALL DIVERT", "GAMES", "CALCULATOR", "REMINDERS", "CLOCK", "PROFILES", "SIM SERVICES"};

                              String[] phoneBookMenu = {"Search","Service Nos.","Add name","Erase",
                              "Edit","Assign tone","Send b'card","Speed dials","Voice tags","Options"};
                               String[] phoneBookOptions = {"Type of view","Memory status"};

          String[] messagesMenu = {"Write messages","Inbox","Outbox","Picture messages",
          "Templates","Smileys","Message settings","Info service","Voice mailbox number","Service command editor"};
          String[] messagesMessagesettings = {"Set", "Common"}; 
          String[] messageMessagesettingsset = {"Message centre number","Message sent as","Message validity"};            
          String[] messageMessagesettingscommon = {"Delivery reports", "Reply via same centre","Character support"};


String[] callRegisterMenu = {"Missed calls", "Received calls", "Dialled numbers",
        "Erase recent call lists", "Show call duration", "Show call costs", "Call cost settings", "Prepaid credit"};
String[] callRegisterShowCallDuration = {"Last call duration", "All calls' duration", "Receive calls' duration",
               "Dialled calls' duration", "Clear timers"}; 
String[] callRegisterShowCallCosts = {"Last call cost", "All calls' cost", "Clear counters"};
String[] callRegisterCallCostSettings = {"Call cost limit", "Show cost in"};


                            String[] tonesMenu = {"Ringing tone","Ringing volume","Incoming call alert","Composer",
                            "Message alert tone","Warming and game tones","Vibration alert","Screen saver"};

      String[] settingsMenu = {"Call settings","Phone settings","Security settings","Restore factory settings"};
               String[] settingsCallSettings = {"Automatic redial", "Speed dialling", "Call waiting options",
                                            "Own number sending", "Phone line in use", "Automatic answer"};
               String[] settingsPhoneSettings = {"Language", "Cell info display","Welcome note", 
                                          "Network selection", "Lights","Confirm SIM service actions"};
               String[] settingsSecuritySettings = {"PIN code request", "Call barring service",
                             "Fixed dialling","Closed user group","Phone security","Change access codes"};

 String[] clockMenu = {"Alarm clock", "Clock settings", "Date setting","Stopwatch",
                        "Counterdown timer","Auto update of date and time"};

              System.out.println("    NOKIA PHONE MENU"); 
               

              for (int i = 1; i <= menus.length; i++) {
                System.out.println(i + " " + menus[i]);
              }


             System.out.println("      CHOOSE FROM THE MENU ONLY");
             String pick = input.nextLine();

            if (pick.equals("PHONE BOOK")) {
               for (String item : phoneBookMenu) {
               System.out.println(item);}
               }
           else if (pick.equals("MESSAGES")) {
               for (String item : messagesMenu) {
               System.out.println(item);}
               }
           else if (pick.equals("CALL REGISTER")) {
               for (String item : callRegisterMenu) {
               System.out.println(item);}
               }
            else if (pick.equals("TONES")) {
               for (String item : tonesMenu) {
               System.out.println(item);}
               }
            else if (pick.equals("SETTINGS")) {
               for (String item : settingsMenu) {
               System.out.println(item);}
               }
            else if (pick.equals("CLOCK")) {
               for (String item : clockMenu) {
               System.out.println(item);}
               }
            else {
               System.out.print("Invalid choice! Please pick from the list.");
            }



 }
}
